### PR TITLE
Fix #2129, notify user when full page is cut off

### LINKED ISSUE #2129: Notify user when Full Page is cut off
We have a 5000px limit to images, even Full Page.  People hit that and then are confused why the images are limited.  We should tell them.

### PR DESCRIPTION
**Pull request by ianb**
Adds a new captureType, fullPageTruncated

Looks like this:

image

### CODE REVIEW

**Comment by ianb** [comment]
New style:

image

**Comment by ianb** [comment]
Or with silly scales:

image

**Comment by ianb** [comment]
new font size:

image

File: static/css/inline-selection.scss [hunk 1]
```diff
@@ -402,6 +402,20 @@ $very-light-grey: #ededed;
   width: 400px;
 }
 
+#imageCroppedWarning {
+  background: rgba(0,0,0,.8);
```
Comment (6a68) [inline static/css/inline-selection.scss]: Could you specify the opacity separately? Seems to make Windows High Contrast mode behave in a more reasonable way

File: shared/shot.js [hunk 2]
```diff
@@ -707,7 +707,7 @@ class _Clip {
     }
     assert(checkObject(image, ["url"], ["dimensions", "text", "location", "captureType", "type"]), "Bad attrs for Clip Image:", Object.keys(image));
     assert(isValidClipImageUrl(image.url), "Bad Clip image URL:", image.url);
-    assert(image.captureType == "madeSelection" || image.captureType == "selection" || image.captureType == "visible" || image.captureType == "auto" || image.captureType == "fullPage" || !image.captureType, "Bad image.captureType:", image.captureType);
+    assert(image.captureType == "madeSelection" || image.captureType == "selection" || image.captureType == "visible" || image.captureType == "auto" || image.captureType == "fullPage" || image.captureType == "fullPageTruncated" || !image.captureType, "Bad image.captureType:", image.captureType);
```
Comment (6a68) [inline shared/shot.js]: this line is looking awfully long

File: locales/en-US/webextension.properties [hunk 3]
```diff
@@ -16,6 +16,8 @@ notificationLinkCopiedTitle = Link Copied
 # modifier key used in the shortcut (do not translate it): for example, Ctrl-V
 # on Windows systems.
 notificationLinkCopiedDetails = The link to your shot has been copied to the clipboard. Press {meta_key}-V to paste.
+# This is displayed as a warning when a Full Page save will be cropped
```
Comment (6a68) [inline locales/en-US/webextension.properties]: Could also mention that {pixels} is a placeholder for a plural whole number, like 5000.

File: addon/webextension/selector/uicontrol.js [hunk 4]
```diff
@@ -160,22 +160,28 @@ this.uicontrol = (function() {
     },
     onClickFullPage: () => {
       sendEvent("capture-full-page", "selection-button");
+      captureType = 'fullPage';
```
Comment (6a68) [inline addon/webextension/selector/uicontrol.js]: y u no double quotes in this file?

File: static/css/inline-selection.scss [hunk 5]
```diff
@@ -402,6 +402,20 @@ $very-light-grey: #ededed;
...
+  bottom: 0;
+  color: #fff;
+  font-family: -apple-system, BlinkMacSystemFont, "segoe ui", "helvetica neue", helvetica, ubuntu, roboto, noto, arial, sans-serif;
+  font-size: 12px;
+  padding: 10px;
+  position: absolute;
+  text-align: center;
+  width: 100%;
+  z-index: 2;
+  position: absolute;
```
Comment (6a68) [inline static/css/inline-selection.scss]: repeated CSS property

File: locales/en-US/webextension.properties [hunk 6]
```diff
@@ -16,6 +16,8 @@ notificationLinkCopiedTitle = Link Copied
...
+imageCroppedWarning = This image has been cropped to {pixels} pixels.
```
Comment (6a68) [inline locales/en-US/webextension.properties]: Should the string be, "This image _will be_ cropped to {pixels} pixels" ?

Comment (flodolo) [reply locales/en-US/webextension.properties]: If you're using "pixel", you should use a proper plural form. The easy way out is to use px.
http://projectfluent.io/fluent/guide/selectors.html

Also CC @stasm to make sure we don't break anything if you go for proper plural.

Comment (6a68) [reply locales/en-US/webextension.properties]: @flodolo In this case the singular will never be needed, so I'm not sure I see the need for the extra selector syntax

File: static/css/inline-selection.scss (hunk 1 above)
Comment (ianb) [reply static/css/inline-selection.scss]: I don't think you can make the background opacity separate, only the entire element opacity...?

**Comment by 6a68** [comment]
@ianb Also, this might be a good time to bump the limit to 10k pixels (#3538)

File: locales/en-US/webextension.properties (hunk 6 above)
Comment (ianb) [reply locales/en-US/webextension.properties]: @johngruen said "px", but I thought that encouraged a term that a normal human wouldn't know so I switched to pixels.

**Comment by ianb** [comment]
This is for uplift to 57, so I think in a meeting we decided not to mess with anything unnecessarily (like the pixel limit)

File: locales/en-US/webextension.properties (hunk 6 above)
Comment (flodolo) [reply locales/en-US/webextension.properties]: Once again, stop using your English brain to think ;-)

one vs many is not a plural form, it's something that works in a bunch of languages
https://developer.mozilla.org/en-US/docs/Mozilla/Localization/Localization_content_best_practices#Use_proper_plural_forms

**Comment by justfortherec** [comment]
> This is for uplift to 57

This sounds like this change is supposed to be in Firefox 57 but I cannot see the message.

Before filing a bug because of truncated full page screenshots I found this change. I cannot see the message in Firefox 57 on Ubuntu. Should I file a bug or will this be available in a future release?

**Comment by chenba** [comment]
@justfortherec It will be in a future release.

**Pull request merged**